src/Analyzers/CSharp/CodeFixes/UseUTF8StringLiteral/UseUTF8StringLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,82 @@
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseUTF8StringLiteral
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseUTF8StringLiteral), Shared]
+    internal class UseUTF8StringLiteralCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        private const char QuoteCharacter = '"';
+        private const string Suffix = "u8";
+
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public UseUTF8StringLiteralCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.UseUTF8StringLiteralDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(CodeAction.Create(
+                CSharpAnalyzersResources.Use_UTF8_string_literal,
+                c => FixAsync(context.Document, context.Diagnostics[0], c),
+                nameof(CSharpAnalyzersResources.Use_UTF8_string_literal)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            foreach (var diagnostic in diagnostics)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var arrayNode = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+                var stringValue = diagnostic.Properties[UseUTF8StringLiteralDiagnosticAnalyzer.StringValuePropertyName]!;
+
+                editor.ReplaceNode(arrayNode, CreateUTF8String(stringValue).WithTriviaFrom(arrayNode));
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private static SyntaxNode CreateUTF8String(string stringValue)
+        {
+            var literal = SyntaxFactory.Token(
+                    leading: SyntaxTriviaList.Empty,
+                    kind: SyntaxKind.UTF8StringLiteralToken,
+                    text: QuoteCharacter + stringValue + QuoteCharacter + Suffix,
+                    valueText: "",
+                    trailing: SyntaxTriviaList.Empty);
```

Comment:
Is it possible to put the trivia here instead of a separate `WithTriviaFrom` call?